src/code/z_demo.c

```diff
@@ -1,99 +1,1650 @@
+
+    if (((gSaveContext.gameMode == 0) || (gSaveContext.gameMode == 1)) && (gSaveContext.respawnFlag <= 0)) {
+        sp2A = func_800F21CC();
+        if (sp2A != -1) {
+            temp_v0_3 = func_800F2138(sp2A);
+            if (temp_v0_3 != -1) {
+                temp_a3 = ((void)0, globalCtx->csCtx.sceneCsList);
+                if ((temp_a3[temp_v0_3].unk7 != 0xFF) && (gSaveContext.respawnFlag == 0)) {
+                    if (temp_a3[temp_v0_3].unk7 == 0xFE) {
+                        ActorCutscene_Start(sp2A, NULL);
+                        gSaveContext.showTitleCard = false;
+                    } else if (!((1 << (temp_a3[temp_v0_3].unk7 % 8)) &
+                                 gSaveContext.weekEventReg[temp_a3[temp_v0_3].unk7 / 8])) {
+                        gSaveContext.weekEventReg[(temp_a3[temp_v0_3].unk7 / 8)] |= 1 << (temp_a3[temp_v0_3].unk7 % 8);
+                        ActorCutscene_Start(sp2A, NULL);
+                        gSaveContext.showTitleCard = false;
+                    }
+                }
+            } else {
+                ActorCutscene_StartAndSetUnkLinkFields(sp2A, NULL);
+            }
+        }
+    }
+
+    if ((gSaveContext.respawnFlag == 0) || (gSaveContext.respawnFlag == -2)) {
+        sp24 = globalCtx->loadedScene;
+        if ((sp24->titleTextId != 0) && gSaveContext.showTitleCard) {
+            if ((Entrance_GetTransitionFlags(gSaveContext.sceneSetupIndex + gSaveContext.entranceIndex) & 0x4000) !=
+                0) {
+                func_80151A68(globalCtx, sp24->titleTextId);
+            }
+        }
+
+        gSaveContext.showTitleCard = true;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDBE0.s")
+#endif
+
+void nop_800EDDB0(GlobalContext* globalCtx) {
+}
+
+void func_800EDDBC(UNK_TYPE arg0, UNK_TYPE arg1) {
+}
+
+void Cutscene_LoadCutsceneData(GlobalContext* globalCtx, u8 csIndex) {
+    if (dREG(95) == 0) {
+        globalCtx->csCtx.currentCsIndex = csIndex;
+        globalCtx->csCtx.csData = Lib_SegmentedToVirtual(globalCtx->csCtx.sceneCsList[csIndex].segmentedData);
+    }
+
+    gSaveContext.cutsceneTrigger = 1;
+}
+
+/**
+ * Interpolates the actor's position based on the corresponding actor action's position
+ * and the current cutscene frame
+ */
+void Cutscene_ActorTranslate(Actor* actor, GlobalContext* globalCtx, s32 actorActionIndex) {
+    Vec3f start;
+    Vec3f end;
+    CsCmdActorAction* entry = globalCtx->csCtx.actorActions[actorActionIndex];
+    f32 t;
+
+    start.x = entry->startPos.x;
+    start.y = entry->startPos.y;
+    start.z = entry->startPos.z;
+    end.x = entry->endPos.x;
+    end.y = entry->endPos.y;
+    end.z = entry->endPos.z;
+
+    t = Environment_LerpWeight(entry->endFrame, entry->startFrame, globalCtx->csCtx.frames);
+
+    actor->world.pos.x = ((end.x - start.x) * t) + start.x;
+    actor->world.pos.y = ((end.y - start.y) * t) + start.y;
+    actor->world.pos.z = ((end.z - start.z) * t) + start.z;
+}
```

Comment:
```suggestion
    progress = Environment_LerpWeight(entry->endFrame, entry->startFrame, globalCtx->csCtx.frames);

    actor->world.pos.x = ((end.x - start.x) * t) + start.x;
    actor->world.pos.y = ((end.y - start.y) * t) + start.y;
    actor->world.pos.z = ((end.z - start.z) * t) + start.z;
}
```
Just for consistency. Also, If you rearrange it such that that start.x/y/z/ is in the front i.e.:
`start.x + ((end.x - start.x) * t)`
Then you can use the `F32_LERPIMP` macro. It's almost certainly going to break regalloc though, but occasionally it doesn't, so it's worth trying